src/app/cities/components/ComparisonModal.tsx

```diff
@@ -0,0 +1,83 @@
+'use client'
+import { CityModel } from "@core/domain/models"
+import { getCityByPromise } from "@core/main/api/cities"
+import { Button, Modal, ModalBody, ModalContent, ModalFooter, ModalHeader } from "@nextui-org/react"
+import { useQueries } from "@tanstack/react-query"
+import { useCompareCities } from "context/ComparisonContext"
+import { isUndefined } from "ramda-adjunct"
+import { CityOverViewTable } from "./CityOverViewTable"
+
+
+export type ComparisonModalProps = {
+  isOpen: boolean
+  onOpenChange: () => void
+  cities: CityModel[]
+}
+
+export function ComparisonModal(props: ComparisonModalProps) {
+  const { isOpen, onOpenChange, cities } = props
+  const { selectedCities, replaceCityBy } = useCompareCities()
+
+  const allCitiesNames = cities.map(({ name }) => name).sort((a, b) => {
+    if (a > b) {
+      return 1
+    }
+    if (a < b) {
+      return -1
+    }
+    return 0
```

Comment:
Esse sort eh para order a lista de cidades na comparação por ordem alfabética? Tem como mudar essa ordem? Precisa desses 2 `if`s aqui e retornar 0?

Reply:
tu quer dizer o user mudar a ordem?

precisa, 0 eh quando os dois sao iguais ai nao inverte ordem
https://dev.to/banesag/sorting-arrays-of-strings-in-javascript-2g11

da pra fazer um helper pra isso, num refactor depois